cmd/ndm_daemonset/controller/blockdevicestore.go

```diff
@@ -83,52 +86,63 @@ func (c *Controller) CreateBlockDevice(blockDevice apis.BlockDevice) error {
 // UpdateBlockDevice update the BlockDevice resource in etcd
 func (c *Controller) UpdateBlockDevice(blockDevice apis.BlockDevice, oldBlockDevice *apis.BlockDevice) error {
```

Comment:
@akhilerm do we really need the second parameter in this function. From what I can understand, the second parameter `oldBlockDevice` is useful only when we want to change the name or namespace of an existing BD. Is this why we have kept it? I feel changing the name shouldn't be allowed by the update operation.

Reply:
The second parameter was provided because we were fetching the latest blockdevice and passing it along with the new resource that we wanted to update 

Name and namespace change is something that we don't want to do.

Reply:
I see. Since now we are anyways using another variable `tempBD` to store the fetched bd from etcd, I think we can do away with the second parameter?